### PR TITLE
Charm++ Tests: Add a testp target for scalable testing on P pes

### PR DESCRIPTION
**Pull request by nitbhat**

### CODE REVIEW

File: tests/charm++/delegation/Makefile [hunk 1]
```diff
@@ -16,8 +18,8 @@ clean: $(foreach i,$(DIRS),clean-$i)
 $(foreach i,$(DIRS),build-$i):
 	$(MAKE) -C $(subst build-,,$@) all OPTS='$(OPTS)'
 
-$(foreach i,$(DIRS),test-$i):
-	$(MAKE) -C $(subst test-,,$@) test OPTS='$(OPTS)' TESTOPTS='$(TESTOPTS)'
+$(foreach i,$(DIRS),testp-$i):
+	$(MAKE) -C $(subst testp-,,$@) testp OPTS='$(OPTS)' TESTOPTS='$(TESTOPTS)' P='$(P)'
```
Comment (evan-charmworks) [inline tests/charm++/delegation/Makefile:22]: Looks like the `test-*` rule generation got accidentally removed here.

File: tests/charm++/partitions/Makefile [hunk 2]
```diff
@@ -27,6 +27,18 @@ else
...
+testp: all
+ifeq ($(CMK_NO_PARTITIONS),1)
+	echo "Skipping test since build does not support partitions"
+else ifeq ($(CMK_BLUEGENEQ),1)
+	echo "Skipping partitions test on BGQ"
+else
+	$(call run, ./hello +p$(P) $$(( $(P) * 4 )) $$(( $(P) / 2 )) +partitions $$(( $(P) / 2 )) )
+endif
+
+
```
Comment (evan-charmworks) [inline tests/charm++/partitions/Makefile:40]: extra newlines at beginning and end

Comment (nitbhat) [reply tests/charm++/partitions/Makefile]: Fixed.

File: tests/charm++/delegation/Makefile (hunk 1 above)
Comment (nitbhat) [reply tests/charm++/delegation/Makefile]: Fixed.

**Pull request merged**